include/seastar/core/semaphore.hh

```diff
@@ -276,6 +319,7 @@ public:
     ///         \ref semaphore_aborted exception.  If the semaphore was
     ///         \ref broken(), may contain a \ref broken_semaphore exception.
     future<> wait(abort_source& as, size_t nr = 1) noexcept {
+        use();
```

Comment:
What about `try_wait()` should we add it there as well?

Reply:
It never creates a waiter so it's not strictly necessary.

Reply:
Hmm, strike that out.  It's called from `try_get_units` that creates semaphore_units that depends on the semaphore, so it is needed after all.

Reply:
Same goes for consume, being called from `consume_units`...